File: 2002B2/CS/WinForms-ErrorHandling-CustomExceptions/CustomExceptions/InsufficientFundsException.cs
using System;

namespace CustomExceptions
{
	/// <summary>
	/// Summary description for InsufficientFundsException.
	/// </summary>
	public class InsufficientFundsException : System.ApplicationException
	{
		public InsufficientFundsException()
		{
		}
		public InsufficientFundsException(string message): base(message)
		{
		}
		public InsufficientFundsException(string message, Exception inner): base(message, inner)
		{
		}
	}
}

File: 2002B2/VB/WinForms-ErrorHandling-CustomExceptions/CustomExceptions/InsufficientFundsException.vb
'Create a custom exception type called InsufficientFundsException. What makes
' it an exception is that it Inherits from System.ApplicationException
Public Class InsufficientFundsException
    Inherits System.ApplicationException

    'Define the three standard constructors for an ApplicationException
    Public Overloads Sub InsufficientFundsException()

    End Sub

    Public Overloads Sub InsufficientFundsException(ByVal Message As String)

    End Sub

    Public Overloads Sub InsufficientFundsException(ByVal Message As String, ByVal Inner As Exception)

    End Sub
End Class
